content/articles/build-ann-with-keras/index.md

```diff
@@ -0,0 +1,247 @@
+
+### Prerequisites
+
+-	Basics of [Artificial Neural Network](https://www.section.io/engineering-education/introduction-to-neural-networks/)
+-	[Google Colab](https://colab.research.google.com/)
+-	Download the Churn modeling dataset from [Kaggle](https://www.kaggle.com/adammaus/predicting-churn-for-bank-customers)
+
+### Outline
+
+-	[Import Libraries](#import-libraries)
+-	[Data Pre-processing](#data-pre-processing)
+-	[Build and Visualize Model](#build-and-visualize-model)
+-	[Train Model](#train-model)
+-	[Make Predictions on New Customer](#make-predictions-on-new-customer)
+
+### Import Libraries
+
+Most of the libraries we will be using have been pre-installed on Google Colab. So, we import them into our code.
+
+```python
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+```
+**Explain what the code below does and why it is necessary**
+
+```python
+print(tf.__version__)
+```
+
+![Tensorflow Version](/engineering-education/build-ann-with-keras/tf-version.PNG)
+
+Let us load our dataset. If you are running the codes with Google Colab, then upload the dataset first. Click on the folder icon on the left panel.
+
+![click on folder icon](/engineering-education/build-ann-with-keras/upload-dataset.PNG)
+
+Then click on the upload icon.
+
+![Upload_dataset](/engineering-education/build-ann-with-keras/upload-data.PNG)
+
+Go to the directory where the dataset is, in your local computer, and select it. Click Open, to upload the dataset to Colab.
+
+![Select dataset in local directory](/engineering-education/build-ann-with-keras/select-dataset.PNG)
+
+![Uploaded dataset](/engineering-education/build-ann-with-keras/uploaded-dataset.PNG)
+
+**Explain here what the code below does**
+
+```python
+dataset = pd.read_csv('/content/Churn_Modelling.csv')
+dataset.head()
+```
+
+![first 5 rows of dataset](/engineering-education/build-ann-with-keras/dataset.PNG)
+
+Let's now pre-process our data.
+
+### Data Pre-processing
+
+Not all the features in our dataset are helpful. We do not need the row number, customer id, and customer names. These features will not help us predict if the customer will churn. Hence, we can get rid of them. We use the code below to separate the features and the label.
+
+```python
+X = dataset.iloc[:, 3:-1].values
+y = dataset.iloc[:, -1].values
+print(X)
+print(y)
+```
+Here are the features and labels obtained after separation:
+
+![features of dataset](/engineering-education/build-ann-with-keras/features.PNG)
+
+![dataset label](/engineering-education/build-ann-with-keras/labels.PNG)
+
+You notice that there are some categorical variables in our dataset. They are in the geography and gender columns. We have to encode these variables. Since there are two unique variables in the gender column, we label-encode it. Then, we one-hot encode the geography column.
+
```

Comment:
Explain briefly the differences between label-encoding and label encoding. This is beneficial to your readers.